components/camel-seda/src/main/java/org/apache/camel/component/seda/SedaComponent.java

```diff
@@ -189,10 +186,27 @@ public synchronized QueueReference getOrCreateQueue(
         log.debug("Created queue {} with size {}", key, size);
 
         // create and add a new reference queue
-        ref = new QueueReference(queue, size, multipleConsumers);
+        QueueReference ref = new QueueReference(queue, size, multipleConsumers);
         ref.addReference(endpoint);
         getQueues().put(key, ref);
```

Comment:
Maybe we should consider creating the queue and the queue reference in a computeIfAbsent to prevent concurrent creation for the same key